tests/light/functional_tests/filterx/test_filterx_update_metric.py

```diff
@@ -0,0 +1,189 @@
+    assert len(samples) == 1
+    assert int(samples[0].value) == 1
+
+    # non literal
+
+    samples = config.get_prometheus_samples([MetricFilter("syslogng_non_literal", {})])
+    assert len(samples) == 3
+
+    samples = config.get_prometheus_samples([MetricFilter("syslogng_non_literal", {"msg": "msg1", "foo": "foovalue"})])
+    assert len(samples) == 1
+    assert int(samples[0].value) == 2
+
+    samples = config.get_prometheus_samples([MetricFilter("syslogng_non_literal", {"msg": "msg2", "foo": "foovalue"})])
+    assert len(samples) == 1
+    assert int(samples[0].value) == 1
+
+    samples = config.get_prometheus_samples([MetricFilter("syslogng_non_literal", {"msg": "msg3", "foo": "foovalue"})])
+    assert len(samples) == 1
+    assert int(samples[0].value) == 1
+
+    syslog_ng.stop()
+
+
+def test_filterx_update_metric_increment(config, port_allocator, syslog_ng):
+    network_source, file_destination = create_config(
+        config,
+        port_allocator,
+        r"""
+            update_metric("const", increment=3);
+            # TODO: fix int cast from message_value
+            update_metric("expr", increment=int(string($MSG)));
+        """,
+    )
+
+    syslog_ng.start(config)
+    network_source.write_log("3\n2\n1\n0\n")
+    file_destination.read_logs(4)
+
+    samples = config.get_prometheus_samples([MetricFilter("syslogng_const", {})])
+    assert len(samples) == 1
+    assert int(samples[0].value) == 12
+
+    samples = config.get_prometheus_samples([MetricFilter("syslogng_expr", {})])
+    assert len(samples) == 1
+    assert int(samples[0].value) == 6
+
+    syslog_ng.stop()
+
+
+def test_filterx_update_metric_level(config, port_allocator, syslog_ng):
+
+    # stats(level(0));
+
+    network_source, file_destination = create_config(
+        config,
+        port_allocator,
+        r"""
+            update_metric("metric", level=2);
+        """,
+        stats_level=0,
+    )
+
+    syslog_ng.start(config)
+    network_source.write_log("foo\n")
+    file_destination.read_log()
+
+    samples = config.get_prometheus_samples([MetricFilter("syslogng_metric", {})])
+    assert len(samples) == 0
+
+    # stats(level(1));
+
+    network_source, file_destination = create_config(
+        config,
+        port_allocator,
+        r"""
+            update_metric("metric", level=2);
+        """,
+        stats_level=1,
+    )
+
+    syslog_ng.reload(config)
+    network_source.write_log("foo\n")
+    file_destination.read_log()
+
+    # stats(level(2));
+
+    network_source, file_destination = create_config(
+        config,
+        port_allocator,
+        r"""
+            update_metric("metric", level=2);
+        """,
+        stats_level=2,
+    )
+
+    syslog_ng.reload(config)
+    network_source.write_log("foo\n")
+    file_destination.read_log()
+
+    samples = config.get_prometheus_samples([MetricFilter("syslogng_metric", {})])
+    assert len(samples) == 1
+    assert int(samples[0].value) == 1
```

Comment:
I think a `syslog_ng.stop()` might be missing here

Reply:
Done, thanks :)